ionoscloud/resource_server.go

```diff
@@ -648,7 +648,11 @@ func resourceServerCreate(ctx context.Context, d *schema.ResourceData, meta inte
 	}
 
 	// Wait, catching any errors
-	_, errState := cloudapi.GetStateChangeConf(meta, d, apiResponse.Header.Get("Location"), schema.TimeoutCreate).WaitForStateContext(ctx)
+	loc, err := apiResponse.Location()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	_, errState := cloudapi.GetStateChangeConf(meta, d, loc.String(), schema.TimeoutCreate).WaitForStateContext(ctx)
 	if errState != nil {
 		if cloudapi.IsRequestFailed(err) {
```

Comment:
redeclaring err in `loc, err := apiResponse.Location()` might affect this, 
but I am unsure what the point of this is in the first place

previously you couldn't reach this point if `err` was not `nil` after the create request, shouldn't it have been `errState`?

Reply:
Yes, this should be errState.